pywbem_mock/_inmemoryrepository.py

```diff
@@ -0,0 +1,725 @@
+#
+# (C) Copyright 2020 InovaDevelopment.comn
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+#
+# Author: Karl  Schopmeyer <inovadevelopment.com>
+#
+
+"""
+Mock WBEMConnection to allow pywbem users to test the pywbem client without
+requiring a running WBEM server. This package mocks the WBEMConnection
+methods that communicate with WBEMServers and mantain a fake server
+repository that is the source and destination for information to
+be used by the methods mocked.
+
+For the module-level documentation, see mocksupport.rst.
+"""
+
+from __future__ import absolute_import, print_function
+
+from pywbem import CIMError, BaseRepositoryConnection, \
+    CIM_ERR_INVALID_NAMESPACE, CIM_ERR_FAILED, CIM_ERR_INVALID_PARAMETER, \
+    CIM_ERR_NOT_FOUND, CIM_ERR_ALREADY_EXISTS, \
+    CIMInstanceName, CIM_ERR_NAMESPACE_NOT_EMPTY
+
+from pywbem._nocasedict import NocaseDict
+
+from pywbem._utils import _format
+from ._resolvermixin import ResolverMixin
+
+__all__ = ['InMemoryRepository']
+
+
+class InMemoryRepository(BaseRepositoryConnection, ResolverMixin):
+    """
+    Create an adaption of the MOF compiler MOFWBEMConnection class so that to
+    directly use the attributes that represent the repository and implement a
+    modified CreateClass that resolves the new class. This directs the compiler
+    output directly  to the dictionaries for qualifiers, and instances in the
+    FakedWBEMConnection object and replaces the CreateClass with a local method
+    that allows resolving the created class before it is inserted into the
+    repository.
+
+    This class is private to pywbem_mock
+    """
+    def __init__(self, conn):
+        """
+        Parameters:
+
+          conn (BaseRepositoryConnection):
+            The underlying repository connection.
+
+            `None` means that there is no underlying repository and all
+            operations performed through this object will fail.
+        """
+
+        self.conn = conn
+        self.conn_id = conn.conn_id
+        self.class_names = {}
```

Comment:
1. Add a description how the `class_names` attribute is organized.
2. Should it also be a NocaseDict at the namespace level?

Reply:
outdated, withdrawing comment.